### PR TITLE
Provide id string to addAnnotatedType

### PR DESCRIPTION
**Pull request by benjamin-confino**
Replaces the depreciated method `addAnnotatedType(AnnotatedType<?> type)` with `void addAnnotatedType(AnnotatedType<?> type,  String id)`

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build 

fixes #5441

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/flow/cdi/FlowBuilderCDIExtension.java [hunk 1]
```diff
@@ -60,7 +62,7 @@ void beforeBeanDiscovery(
         // can take it into account, and use it later when necessary.
         AnnotatedType<FlowBuilderFactoryBean> flowDiscoveryHelper = 
                         beanManager.createAnnotatedType(FlowBuilderFactoryBean.class);
-        event.addAnnotatedType(flowDiscoveryHelper);
+        event.addAnnotatedType(flowDiscoveryHelper, CDIServiceUtils.getAnnotatedTypeIdentifier(flowDiscoveryHelper, this.getClass()));
```
Comment (tevans78) [inline dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/flow/cdi/FlowBuilderCDIExtension.java]: Need to investigate why this class does not implement WebSphereCDIExtension

File: dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/cdi/view/ViewScopeContextExtension.java [hunk 2]
```diff
@@ -42,9 +44,9 @@ void beforeBeanDiscovery(
         // Register ViewScopeBeanHolder as a bean with CDI annotations, so the system
         // can take it into account, and use it later when necessary.
         AnnotatedType appContext = beanManager.createAnnotatedType(ApplicationContextBean.class);
-        event.addAnnotatedType(appContext);
+        event.addAnnotatedType(appContext, CDIServiceUtils.getAnnotatedTypeIdentifier(appContext, this.getClass()));
```
Comment (tevans78) [inline dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/cdi/view/ViewScopeContextExtension.java]: Need to investigate why this class does not implement WebSphereCDIExtension

File: dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/cdi/dependent/DependentBeanExtension.java [hunk 3]
```diff
@@ -19,15 +19,19 @@
...
 
+import com.ibm.ws.cdi.CDIServiceUtils;
+
 public class DependentBeanExtension implements Extension
 {
     @SuppressWarnings("UnusedDeclaration")
     public void registerAnnotatedTypes(@Observes BeforeBeanDiscovery beforeBeanDiscovery, BeanManager beanManager)
     {
-        beforeBeanDiscovery.addAnnotatedType(beanManager.createAnnotatedType(RequestDependentBeanStorage.class));
+        AnnotatedType at = beanManager.createAnnotatedType(RequestDependentBeanStorage.class);
```
Comment (tevans78) [inline dev/com.ibm.ws.jsf.2.2/src/org/apache/myfaces/cdi/dependent/DependentBeanExtension.java]: Need to investigate why this class does not implement WebSphereCDIExtension

**Pull request merged**